streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -489,8 +496,12 @@ public void updateSubscriptionsFromMetadata(final Set<String> topics) {
     /**
      * @throws TaskMigratedException if committing offsets failed (non-EOS)
      *                               or if the task producer got fenced (EOS)
+     * @return number of committed offsets, or -1 if we are in the middle of a rebalance and cannot commit
      */
     int commitAll() {
+        if (rebalanceInProgress) {
```

Comment:
It's actually fine to call `standby.commit` during a rebalance, I just wasn't sure how that would work with the metrics (it would screw with `commit-latency` to be sometimes recording the latency of active + standby commits, and sometimes just standby). Maybe we could still commit the standbys but pretend like we didn't call commit (ie, not record anything?) Or would that screw up the metrics even more @guozhangwang

Reply:
On a somewhat related note, I notice we don't actually call commit on restoring tasks but we do on standbys. Is there a particular reason for this? What is there to commit for a standby task but not for a restoring?

Reply:
As long as we have separate tags for standby and primary tasks commit, I don't think the metrics will be messed up.

Reply:
I don't think we do though -- currently we just call `commitSensor.record(intervalCommitLatency / (double) committed, now);` where the commitLatency is for committing all tasks